dropwizard-auth/src/main/java/io/dropwizard/auth/PolymorphicAuthDynamicFeature.java

```diff
@@ -0,0 +1,45 @@
+package io.dropwizard.auth;
+
+import com.google.common.collect.ImmutableMap;
+import org.glassfish.jersey.server.model.AnnotatedMethod;
+
+import java.lang.annotation.Annotation;
+import java.security.Principal;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+
+/**
+ * A {@link DynamicFeature} that registers the provided auth filters
+ * to resource methods annotated with the {@link Auth} according to
+ * the type of the annotated method parameter.
+ * <p>Note that this feature only pertains to <i>authentication</i>.
+ * To enable authorization, use {@link AuthDynamicFeature}.</p>
+ */
+public class PolymorphicAuthDynamicFeature<T extends Principal> implements DynamicFeature {
```

Comment:
Could you provide a minimal test case that uses this feature with several auth filters? I would prefer to see tests for every contributed functionality.